test/e2e/common.go

```diff
@@ -125,19 +125,24 @@ func getSha256Hash(filename string) ([]byte, error) {
 	return hash.Sum(nil), nil
 }
 
-func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, clusterProxy framework.ClusterProxy, artifactFolder string, namespace string, intervalsGetter func(spec, key string) []interface{}, clusterName, clusterctlLogFolder string, skipCleanup bool) {
+func DumpSpecResourcesAndCleanup(ctx context.Context, specName string, bootstrapClusterProxy framework.ClusterProxy, targetClusterProxy framework.ClusterProxy, artifactFolder string, namespace string, intervalsGetter func(spec, key string) []interface{}, clusterName, clusterctlLogFolder string, skipCleanup bool) {
```

Comment:
I'm a bit torn about the change here since it implies that there will always be exactly one target cluster.
What if we have tests with multiple workload clusters at some point?
Not a huge issue, but something to think about. Perhaps we could instead detect all workload clusters automatically?

Reply:
I agree, I can open a separate PR with this change. Can we merge this one without this change? I will add the to-do to the function.

Reply:
Yes I'm fine to merge like this and work on improving it later :slightly_smiling_face: